### ISSUE TITLE
Gjs-WARNING: The GObject property St.ScrollView.[hv]scroll is deprecated (in GNOME 46)

### ISSUE THREAD

**Issue by vanvugt** [micheleg/dash-to-dock#2159]
In GNOME 46:

```
(gnome-shell:18994): Gjs-WARNING **: 13:53:18.466: The GObject property St.ScrollView.vscroll is deprecated.
0 ensureActorVisibleInScrollView() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/dash.js":1143:38]
1 _createAppItem/<() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/dash.js":569:46]
2 _generateAccessors/propdesc.set() ["resource:///org/gnome/gjs/modules/core/_common.js":38:25]
3 _updateFocusState() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/appIcons.js":328:8]
4 _init/<() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/appIcons.js":921:74]
5 popModal() ["resource:///org/gnome/shell/ui/main.js":739:16]
6 _syncGrab() ["resource:///org/gnome/shell/ui/overview.js":526:21]
7 _hideDone() ["resource:///org/gnome/shell/ui/overview.js":647:13]
8 _animateNotVisible/<() ["resource:///org/gnome/shell/ui/overview.js":627:54]
9 onStopped() ["resource:///org/gnome/shell/ui/overviewControls.js":757:20]
10 _makeEaseCallback/<() ["resource:///org/gnome/shell/ui/environment.js":65:21]
11 _easeActorProperty/<() ["resource:///org/gnome/shell/ui/environment.js":232:59]
12 anonymous() ["resource:///org/gnome/shell/ui/init.js":21:19]


(gnome-shell:18994): Gjs-WARNING **: 13:53:18.466: The GObject property St.ScrollView.hscroll is deprecated.
0 ensureActorVisibleInScrollView() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/dash.js":1144:38]
1 _createAppItem/<() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/dash.js":569:46]
2 _generateAccessors/propdesc.set() ["resource:///org/gnome/gjs/modules/core/_common.js":38:25]
3 _updateFocusState() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/appIcons.js":328:8]
4 _init/<() ["file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/appIcons.js":921:74]
5 popModal() ["resource:///org/gnome/shell/ui/main.js":739:16]
6 _syncGrab() ["resource:///org/gnome/shell/ui/overview.js":526:21]
7 _hideDone() ["resource:///org/gnome/shell/ui/overview.js":647:13]
8 _animateNotVisible/<() ["resource:///org/gnome/shell/ui/overview.js":627:54]
9 onStopped() ["resource:///org/gnome/shell/ui/overviewControls.js":757:20]
10 _makeEaseCallback/<() ["resource:///org/gnome/shell/ui/environment.js":65:21]
11 _easeActorProperty/<() ["resource:///org/gnome/shell/ui/environment.js":232:59]
12 anonymous() ["resource:///org/gnome/shell/ui/init.js":21:19]
```

**Comment by sergio-costas** [comment]
This is odd... neither the documentation nor the Gnome Shell source code do mark those properties as deprecated...

![imagen](https://github.com/micheleg/dash-to-dock/assets/105285003/2bc03f80-dac0-4853-8713-5f2b7f0b5a70)

![imagen](https://github.com/micheleg/dash-to-dock/assets/105285003/36151bd7-9e5b-4e93-a33b-4acfa0e40450)

**Comment by vanvugt** [comment]
```
$ grep -B4 DEPRECATED *
...
--
st-scroll-view.c-    g_param_spec_object ("hscroll",
st-scroll-view.c-                         "StScrollBar",
st-scroll-view.c-                         "Horizontal scroll indicator",
st-scroll-view.c-                         ST_TYPE_SCROLL_BAR,
st-scroll-view.c:                         ST_PARAM_READABLE | G_PARAM_DEPRECATED);
--
st-scroll-view.c-    g_param_spec_object ("vscroll",
st-scroll-view.c-                         "StScrollBar",
st-scroll-view.c-                         "Vertical scroll indicator",
st-scroll-view.c-                         ST_TYPE_SCROLL_BAR,
st-scroll-view.c:                         ST_PARAM_READABLE | G_PARAM_DEPRECATED);
--
...
```

**Comment by sergio-costas** [comment]
Oh! I was searching for something in the comments... Now it makes sense.

**Comment by sergio-costas** [comment]
(also, I searched in lowercase 🤦🏾‍♂️)

**Comment by K1ngjulien** [comment]
Am I correct in seeing that this is the only place this occurs?:

https://github.com/micheleg/dash-to-dock/blob/936b8663d8f55f43e2784f844ec61264e4a697fb/dash.js#L1142-L1146

From what I can tell, it should be possible to just access `scrollView.vAdjustment` and `scrollView.hAdjustment` directly instead of going through `vscroll` and `hscroll`.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3020/diffs?commit_id=d243a9ba9320971d5ce8524ff8355b24606d9420

That seems to by why they got deprecated it in the first place:

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3020

**Comment by K1ngjulien** [comment]
@vanvugt can you check the PR? that should be the last issue for the gnome 46 release :)

**Comment by sergio-costas** [comment]
@K1ngjulien You are right. I left a comment in your PR, but in overall, it looks good.

**Comment by vanvugt** [comment]
Closed in #2185.
